module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,40 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressOneLine(const std::array<uint8_t, width> &line)
+{
+    std::vector<std::pair<uint8_t, uint8_t>> vect;
```

Comment:
You can reserve memory for vector and at end call `shrink_to_fit` You avoid few allocations which are slow.